src/Main.elm

```diff
@@ -2,21 +2,147 @@ module Main exposing (main)
 
 import Browser
 import Data
+import Dict exposing (Dict)
 import Html exposing (Html)
 import Html.Attributes as Attrs
+import Html.Events as Events
+import Json.Decode as Decode
+import Json.Decode.Pipeline as Decode
+import Modules.Contact as Contact
+import Modules.Settings as Settings exposing (Settings)
+import Modules.Tags as Tags
+
+
+
+---- PROGRAM ----
+
+
+main : Program () Model Msg
+main =
+    Browser.application
+        { view =
+            \model ->
+                { title = "Scrive elm challenge task"
+                , body = [ view model ]
+                }
+        , init = \_ _ _ -> init
+        , update = update
+        , subscriptions = always Sub.none
+        , onUrlRequest = always NoOp
+        , onUrlChange = always NoOp
+        }
 
 
 
 ---- MODEL ----
 
 
 type alias Model =
-    {}
+    { userGroup : UserGroup
+    , currentForm : Maybe Form
+    , contactForm : Contact.Model
+    , settingsForm : Settings.Model
+    , tagsForm : Tags.Model
+    }
+
+
+type Form
+    = SettingsForm
+    | ContactForm
+    | TagsForm
+
+
+
+-- User Group
+
+
+type alias UserGroup =
+    { id : String
+    , parentId : String
+    , name : String
+    , children : List Children
+    , settings : Settings
+    , contactDetails : Contact.Details
+    , tags : Dict String String
+    }
+
+
+emptyUserGroup : UserGroup
+emptyUserGroup =
+    { id = ""
+    , parentId = ""
```

Comment:
In the `model`, `newTagError` is a `Maybe String`, ie, the "no error" condition is modeled by `Nothing`.
Here instead, the "no parent" condition is modeled by `""`.

There isn't really a correct way to do this, but I would like to hear from the candidate about why they choose different approaches in the two instances.